scripts/bootloader/provision.py

```diff
@@ -13,27 +13,64 @@
 from hashlib import sha256
 import os
 
+# Size of LCS storage in OTP in bytes
+LCS_STATE_SIZE = 0x8
 
-def generate_provision_hex_file(s0_address, s1_address, hashes, provision_address, output, max_size,
-                                num_counter_slots_version):
+def add_hw_counters(provision_data, nsib_counter_slots, mcuboot_counters_num, mcuboot_counters_slots):
+    all_counters_num = mcuboot_counters_num + 1 if nsib_counter_slots > 0 else mcuboot_counters_num
+
+    if all_counters_num == 0:
+        return provision_data
+
+    if nsib_counter_slots > 0 and (nsib_counter_slots % 2 == 1):
+            nsib_counter_slots+= 1
+            print(f'Monotonic counter slots for NSIB rounded up to {nsib_counter_slots}')
+
+    if mcuboot_counters_slots > 0 and (mcuboot_counters_slots % 2 == 1):
+            mcuboot_counters_slots+= 1
+            print(f'Monotonic counter slots for MCUBOOT rounded up to {mcuboot_counters_slots}')
+
+    provision_data += struct.pack('H', 1) # Type "counter collection"
+    provision_data += struct.pack('H', all_counters_num)
+
+    if nsib_counter_slots > 0:
+            provision_data += struct.pack('H', 1) # counter description
+            provision_data += struct.pack('H', nsib_counter_slots)
+            provision_data += bytes(2 * nsib_counter_slots * [0xFF])
+
+    for i in range(0, mcuboot_counters_num):
+        provision_data += struct.pack('H', i + 2) # MCUBOOT counters ID start from 2
+        provision_data += struct.pack('H', mcuboot_counters_slots)
+        provision_data += bytes(2 * mcuboot_counters_slots* [0xFF])
+
+    return provision_data
+
+def generate_mcuboot_only_provision_hex_file(provision_address, output, max_size, mcuboot_counters_num, mcuboot_counters_slots):
     # Add addresses
-    provision_data = struct.pack('III', s0_address, s1_address, len(hashes))
+    provision_data = add_hw_counters(bytes(), 0, mcuboot_counters_num, mcuboot_counters_slots)
+
+    assert len(provision_data) <= max_size, """Provisioning data doesn't fit. Reduce the number of counter slots and try again."""
+
+    ih = IntelHex()
+    ih.frombytes(provision_data, offset=provision_address)
+    ih.write_hex_file(output)
```

Comment:
Could be wrapped into its own function, as it is basically the same tan in `generate_provision_hex_file`